li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/largemessage/ConsumerRecordsProcessor.java

```diff
@@ -433,6 +434,24 @@ public long getRecordsSkipped() {
     return recordsSkipped;
   }
 
+  /**
+   * Gives an idea about how far behind the safe offset of a partition is from the watermark. Hence, the
+   * offset-watermark span or a partition is computed as (watermark minus safeoffset).
+   * This metric is a sum of
+   * {@link com.linkedin.kafka.clients.largemessage.DeliveredMessageOffsetTracker#offsetWatermarkSpan} for all
+   * assigned partitions that have currently undelivered large messages in the consumer. It is computed on query of
+   * the metric each time.
+   */
+  public long getConsumerOffsetWatermarkSpan() {
+    long span = 0;
+    // mem copy to avoid ConcurrentModificationException, in case known partition set changes
+    final Set<TopicPartition> knownPartitions = new HashSet<>(knownPartitions());
```

Comment:
The HashSet constructor calls addAll which does this:
```java
 public boolean addAll(Collection<? extends E> c) {
        boolean modified = false;
        for (E e : c)
            if (add(e))
                modified = true;
        return modified;
    }
```

This can still get CME because HashSet<>(Collection<>) will use an iterator to make a copy.   There needs to be some kind of synchronization in order to avoid this or use some kind of concurrent data structure.